guides/20240820_guide_template copy.md

```diff
@@ -0,0 +1,268 @@
+
+Using Hugging Face as an example highlights the importance of having a pre-configured environment tailored for ML and NLP tasks. This feature can save developers time and reduce the potential for setup errors, enabling them to focus more on development and less on configuration.
+
+### Prerequisites
+
+o follow along with this guide, you should have a basic understanding of[Python](definitions\20240820_defintion_python.md), [shell scripting](definitions\20240820_definition_shell_scripting.md) and [containerization](definitions/20240819_definition_containerization.md).
+
+You will also need the following tools installed: [Docker](https://www.docker.com/),  [Visual Studio Code](https://code.visualstudio.com/), and the [Remote - Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers).
+
+- **Docker**: Used for containerizing applications, ensuring consistency across different environments.
+- **Visual Studio Code**: An Integrated Development Environment (IDE) that supports Devcontainer development.
+- **Remote - Containers Extension**: A VS Code extension that allows you to open any folder inside a container and take advantage of Visual Studio Code’s full feature set.
+
+### TL;DR
+
+- **Clone and Prepare**: Fork and clone the feature-starter repository.
+- **Create Feature Directory**: Set up the `huggingface` directory within the `src` folder.
+- **Configure and Implement**: Define metadata in `devcontainer-feature.json` and implement the feature installation script.
+- **Test and Publish**: Test the feature locally and publish it using GitHub Actions.
+
+## Step 1: Preparations
+
+### Fork the Starter Repository
+
+To create your custom Devcontainer feature, start by forking the [feature-starter repository](https://github.com/devcontainers/feature-starter) on GitHub. This repository provides a solid foundation for building new Devcontainer features.
+
+- **Step 1.1**: Fork the Repository
+
+Click on the "Fork" button at the top right of the repository page to create a local copy under your GitHub account.
+
+- **Step 1.2**: Clone this repository:
+
+Once forked, clone the repository to your local machine using the following commands:
+
+```bash
+git clone https://github.com/your-username/feature-starter.git
+cd feature-starter
+```
+
+- **Step 1.3**: Open the project in VS Code:
+
+```bash
+code .
+```
+
+- **Step 1.4**: When prompted, click "Reopen in Container" to develop inside a container with the necessary tools installed.
+
+**Note:** _[Ensure your development environment is set up with Docker, Visual Studio Code and the Remote - Containers extension]_
+
+## Step 2: Main Process
+
+### Create a New Feature Directory
+
+Inside the `src` directory of your forked repository, create a new folder named `huggingface`. This folder will contain all the necessary configuration files and scripts for the Hugging Face feature.
```

Comment:
A self-review for myself:

Can be added devcontainer cli installation cd and as well as a file structure of how devcontainer expects ?

Something like this"
```
├── src/
│   ├── huggingface/
│   │   ├── devcontainer-feature.json
│   │   ├── install.sh
│   │   ├── test/
│   │   │   └── test.sh
```